packages/playback-core/src/tracks.ts

```diff
@@ -127,6 +124,19 @@ function createTextTrack(
     trackEl.id = id;
   }
   trackEl.track.mode = 'disabled';
+  // Add data attribute to identify tracks that should be removed when switching sources/destroying hls.js instance.
+  trackEl.setAttribute('data-removeondestroy', '');
```

Comment:
can we remember these tracks without setting any attributes? would a weakmap work?

Reply:
> can we remember these tracks without setting any attributes? would a weakmap work?

@luwes We can, but I actually prefer using the attribute for:
1. Simplicity (we only add it on creation and use it via query selectors to identify items to remove)
2. Visibility (programmatic and devs trying to e.g. debug)
3. Avoiding module level state (harder to test/debug at a unit level since we now have modules whose state changes over time once loaded)
4. Applying a valid use case for data attributes (a data-driven identifier associated with an element instance)
5. Versatility (advanced users can easily add the attribute if they want to guarantee the track gets removed when the src changes without having to build any architecture)

If you feel strongly about the change and see particular advantages to it, we can get a gut check from the rest of the team and implement in whichever way the team as a whole prefers.

Reply:
cool I'm ok with it, my preference for keeping the data- attribute out was more for not adding special HTML to the document.

I would argue against point 5 and best to keep it out of their control because it can already be done with the API methods.